### PR TITLE
fix: add Dart `required` keyword

### PR DESCRIPTION
**Pull request by mhmdanas**
Not sure if this should be `feat` or `fix`.

### CODE REVIEW

File: lexers/embedded/dart.xml [hunk 1]
```diff
@@ -97,7 +97,7 @@
       <rule pattern="\b(assert|break|case|catch|continue|default|do|else|finally|for|if|in|is|new|return|super|switch|this|throw|try|while)\b">
         <token type="Keyword"/>
       </rule>
-      <rule pattern="\b(abstract|async|await|const|extends|factory|final|get|implements|native|operator|set|static|sync|typedef|var|with|yield)\b">
+      <rule pattern="\b(abstract|async|await|const|extends|factory|final|get|implements|native|operator|required|set|static|sync|typedef|var|with|yield)\b">
```
Comment (mhmdanas) [inline lexers/embedded/dart.xml:100]: I don't know if I should add `required` here or to the rule above, what do you think?

File: lexers/testdata/dart.actual [hunk 2]
```diff
@@ -3,8 +3,8 @@ import 'dart:html';
 import 'dart:math' show Random;
 
 // We changed 5 lines of code to make this sample nicer on
-// the web (so that the execution waits for animation frame, 
-// the number gets updated in the DOM, and the program ends 
+// the web (so that the execution waits for animation frame,
+// the number gets updated in the DOM, and the program ends
```
Comment (mhmdanas) [inline lexers/testdata/dart.actual:7]: If you don't want these formatting changes in this PR I can remove them.

File: lexers/embedded/dart.xml (hunk 1 above)
Comment (alecthomas) [reply lexers/embedded/dart.xml:100]: Here seems fine to me.

**Comment by alecthomas** [comment]
Thanks!

**Comment by mhmdanas** [comment]
You're welcome!

**Pull request merged**